src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -876,15 +804,27 @@ inline void write_output_segments(Segments& output_segments, const std::shared_p
           // Get the row ids that are referenced
           auto new_pos_list = std::make_shared<PosList>(pos_list->size());
           auto new_pos_list_iter = new_pos_list->begin();
+          auto common_chunk_id = std::optional<ChunkID>{};
           for (const auto& row : *pos_list) {
             if (row.chunk_offset == INVALID_CHUNK_OFFSET) {
               *new_pos_list_iter = row;
+              common_chunk_id = INVALID_CHUNK_ID;
             } else {
               const auto& referenced_pos_list = *(*input_table_pos_lists)[row.chunk_id];
               *new_pos_list_iter = referenced_pos_list[row.chunk_offset];
+
+              const auto referenced_chunk_id = referenced_pos_list[row.chunk_offset].chunk_id;
```

Comment:
Add a comment that this part is only there to set the flag for single chunk?

Reply:
"Check if the current row matches the ChunkIDs that we have seen in previous rows"